From 490eabe5eed2fccd8d174fc07cabbb5e487f09c9 Mon Sep 17 00:00:00 2001
From: NLObP <38990417+NL0bP@users.noreply.github.com>
Date: Tue, 13 Aug 2024 22:08:25 +0300
Subject: [PATCH] Merge errors fixed;

---
 AAEmu.Game/GameData/AchievementGameData.cs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/AAEmu.Game/GameData/AchievementGameData.cs b/AAEmu.Game/GameData/AchievementGameData.cs
index 0d8971f82..be37fd0c9 100644
--- a/AAEmu.Game/GameData/AchievementGameData.cs
+++ b/AAEmu.Game/GameData/AchievementGameData.cs
@@ -22,7 +22,7 @@ public class AchievementGameData : Singleton<AchievementGameData>, IGameDataLoad
     private Dictionary<uint, List<AchievementObjectives>> _achievementObjectives;
     private Dictionary<uint, List<PreCompletedAchievements>> _preCompletedAchievements;
 
-    public void Load(SqliteConnection connection)
+    public void Load(SqliteConnection connection, SqliteConnection connection2)
     {
         _charRecords = new Dictionary<uint, CharRecords>();
         _achievements = new Dictionary<uint, Achievements>();
