From fe3b318a34299ddf74ec4edadff32a3627189a35 Mon Sep 17 00:00:00 2001
From: Yanlongli <ahlyl94@gmail.com>
Date: Fri, 16 Aug 2024 13:34:51 +0800
Subject: [PATCH] [Game] Check recipients when sending mails (#1080)

---
 AAEmu.Game/Core/Packets/C2G/CSSendMailPacket.cs | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/AAEmu.Game/Core/Packets/C2G/CSSendMailPacket.cs b/AAEmu.Game/Core/Packets/C2G/CSSendMailPacket.cs
index 12a42d6fb..d667fb7a4 100644
--- a/AAEmu.Game/Core/Packets/C2G/CSSendMailPacket.cs
+++ b/AAEmu.Game/Core/Packets/C2G/CSSendMailPacket.cs
@@ -2,6 +2,8 @@
 using System.Linq;
 
 using AAEmu.Commons.Network;
+using AAEmu.Game.Core.Managers;
+using AAEmu.Game.Core.Managers.UnitManagers;
 using AAEmu.Game.Core.Managers.World;
 using AAEmu.Game.Core.Network.Game;
 using AAEmu.Game.Models.Game;
@@ -47,6 +49,13 @@ public override void Read(PacketStream stream)
 
         // Validate if we are near a MailBox
         bool mailCheckOK;
+
+        if (string.IsNullOrWhiteSpace(receiverCharName) || NameManager.Instance.GetCharacterId(receiverCharName) == 0)
+        {
+            Connection.ActiveChar.SendErrorMessage(ErrorMessageType.MailReceiverNotFound);
+            return;
+        }
+        
         if (doodad != null)
         {
             // Cannot rely on doodad GroupID being "Other - Mailboxes (6)", as some of the mailboxes belong to other groups (e.g. "Housing - Furniture").
